setup.py

```diff
@@ -28,9 +28,12 @@
 # SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 ###############################################################################
 
+import logging
 from setuptools import setup
 from setuptools import find_packages
 from os import path, walk
+from cocotb.build_libs import get_ext, build_ext
```

Comment:
I don't think this self-import is particularly safe - it will likely fail if installed from anywhere but `.`.

Numpy does something similar with `numpy.distutils`, but wraps the work in:
```
    src_path = os.path.dirname(os.path.abspath(__file__))
    old_path = os.getcwd()
    os.chdir(src_path)
    sys.path.insert(0, src_path)
    try:
        setup(...)
    finally:
        del sys.path[0]
        os.chdir(old_path)
```

Reply:
Seems to work for me as is (tired different directory and form page).

Reply:
So `cd /tmp; pip uninstall cocotb; pip install full/path/to/cocotb` works?

Reply:
I think this might break in virtualenvs, I remember running into this problem with another package.

Reply:
>So cd /tmp; pip uninstall cocotb; pip install full/path/to/cocotb works?
Yes

This also `pip install -v http://github.com/themperek/cocotb/archive/build_on_setup.zip`

Reply:
Ok, I now remember why this is a bad idea - it prevents cocotb ever taking external dependencies, because you import cocotb before you try and install its dependencies

Reply:
Do you envisage users ever importing `build_libs`? If not, then I'd propose we move this stuff to `setup_build_libs.py` or similar, and import it from there (and not install it with cocotb).

Reply:
It seems some requested to have a separate command like "cocotb-build-libs": https://github.com/cocotb/cocotb/issues/1062#issuecomment-519586473

Reply:
Can we keep this "not ugly" and fix in case someone spots an issue?

Reply:
As far as I can read this import should be fine if there are no extra dependencies.

Reply:
How about:
* rename to `cocotb._build_libs`, to stop anyone depending on this directly (yet)
* Adding a comment here saying something like
  ```
  # note: cocotb is not installed properly yet, but we can import it anyway
  # because it's in the currently directory. We'll need to change this if we add to
  # `install_requires`.
  ```